teamcity: failed test: TestCreateStatsLivenessWithRestart

The following tests appear to have failed on master (test): TestCreateStatsLivenessWithRestart

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestCreateStatsLivenessWithRestart).

[#1343894](https://teamcity.cockroachdb.com/viewLog.html?buildId=1343894):


```
TestCreateStatsLivenessWithRestart
...617 18:23:48.365130 39366 server/server.go:1593  [n1] starting grpc/postgres server at 127.0.0.1:36199
I190617 18:23:48.365155 39366 server/server.go:1594  [n1] advertising CockroachDB node at 127.0.0.1:36199
I190617 18:23:48.448507 39631 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I190617 18:23:48.487334 39723 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.1-4 User:root}
I190617 18:23:48.522193 39810 sql/event_log.go:132  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
E190617 18:23:48.535159 39705 jobs/registry.go:325  error while adopting jobs: adopt-job: column "progress" does not exist
I190617 18:23:48.553634 39950 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:995fa611-6947-4da5-b042-181fa7871a32 User:root}
I190617 18:23:48.561815 40029 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I190617 18:23:48.566578 40034 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I190617 18:23:48.577930 39366 server/server.go:1662  [n1] done ensuring all necessary migrations have run
I190617 18:23:48.577971 39366 server/server.go:1665  [n1] serving sql connections
I190617 18:23:48.604561 39427 server/server_update.go:70  [n1] no need to upgrade, cluster already at the newest version
I190617 18:23:48.615602 39430 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:36199 Attrs: Locality:region=test,dc=dc1 ServerVersion:19.1-4 BuildTag:v19.2.0-alpha.20190606-150-g6c7bea6 StartedAt:1560795828338896888 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:083be022-48ba-4f32-a7c1-995fad3681c4 StartedAt:1560795828338896888 LastUp:1560795828338896888}
I190617 18:23:48.634888 40201 sql/event_log.go:132  [n1,client=127.0.0.1:43810,user=root] Event: "create_database", target: 52, info: {DatabaseName:liveness Statement:CREATE DATABASE liveness User:root}
I190617 18:23:48.643047 40201 sql/event_log.go:132  [n1,client=127.0.0.1:43810,user=root] Event: "create_table", target: 53, info: {TableName:liveness.public.t Statement:CREATE TABLE liveness.public.t (i INT8 PRIMARY KEY) User:root}
I190617 18:23:48.650629 40310 storage/replica_command.go:290  [n1,split,s1,r20/1:/{Table/24-Max}] initiating a split of this range at key /Table/53 [r21] (zone config)
W190617 18:23:48.833944 39703 jobs/registry.go:780  job 461129084783067137: canceling due to liveness failure
I190617 18:23:49.007044 40398 util/stop/stopper.go:544  quiescing; tasks left:
9      [async] storage.IntentResolver: cleanup txn records
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
W190617 18:23:49.007518 39441 storage/intentresolver/intent_resolver.go:820  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/24/RangeDescriptor: node unavailable; try another peer
I190617 18:23:49.008715 40398 util/stop/stopper.go:544  quiescing; tasks left:
9      [async] storage.IntentResolver: cleanup txn records
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I190617 18:23:49.008856 40398 util/stop/stopper.go:544  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
    create_stats_job_test.go:213: expecting rescheduled job to have a later modification time: 1560795828000000 vs 1560795828705777




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/jobs/jobs.go
+++ b/pkg/jobs/jobs.go
@@ -136,7 +136,7 @@ func (j *Job) Started(ctx context.Context) error {
 			return nil
 		}
 		ju.UpdateStatus(StatusRunning)
-		md.Payload.StartedMicros = timeutil.ToUnixMicros(timeutil.Now())
+		md.Payload.StartedMicros = timeutil.ToUnixMicros(j.registry.clock.Now().GoTime())
 		ju.UpdatePayload(md.Payload)
 		return nil
 	})
@@ -325,7 +325,7 @@ func (j *Job) canceled(ctx context.Context, fn func(context.Context, *client.Txn
 			}
 		}
 		ju.UpdateStatus(StatusCanceled)
-		md.Payload.FinishedMicros = timeutil.ToUnixMicros(timeutil.Now())
+		md.Payload.FinishedMicros = timeutil.ToUnixMicros(j.registry.clock.Now().GoTime())
 		ju.UpdatePayload(md.Payload)
 		return nil
 	})
@@ -349,7 +349,7 @@ func (j *Job) Failed(
 		}
 		ju.UpdateStatus(StatusFailed)
 		md.Payload.Error = err.Error()
-		md.Payload.FinishedMicros = timeutil.ToUnixMicros(timeutil.Now())
+		md.Payload.FinishedMicros = timeutil.ToUnixMicros(j.registry.clock.Now().GoTime())
 		ju.UpdatePayload(md.Payload)
 		return nil
 	})
@@ -367,7 +367,7 @@ func (j *Job) Succeeded(ctx context.Context, fn func(context.Context, *client.Tx
 			return err
 		}
 		ju.UpdateStatus(StatusSucceeded)
-		md.Payload.FinishedMicros = timeutil.ToUnixMicros(timeutil.Now())
+		md.Payload.FinishedMicros = timeutil.ToUnixMicros(j.registry.clock.Now().GoTime())
 		ju.UpdatePayload(md.Payload)
 		md.Progress.Progress = &jobspb.Progress_FractionCompleted{
 			FractionCompleted: 1.0,

--- a/pkg/jobs/update.go
+++ b/pkg/jobs/update.go
@@ -173,7 +173,7 @@ func (j *Job) Update(ctx context.Context, updateFn UpdateFn) error {
 
 		if ju.md.Progress != nil {
 			progress = ju.md.Progress
-			progress.ModifiedMicros = timeutil.ToUnixMicros(timeutil.Now())
+			progress.ModifiedMicros = timeutil.ToUnixMicros(txn.OrigTimestamp().GoTime())
 			progressBytes, err := protoutil.Marshal(progress)
 			if err != nil {
 				return err

